twake/frontend/src/app/scenes/Apps/Messages/Message/Parts/Reactions.tsx

```diff
@@ -2,75 +2,77 @@ import React from 'react';
 import User from 'services/user/UserService';
 import Collections from 'app/services/Depreciated/Collections/Collections.js';
 import 'moment-timezone';
-import Tooltip from 'components/Tooltip/Tooltip.js';
 import MessagesService from 'services/Apps/Messages/Messages.js';
 import Emojione from 'components/Emojione/Emojione';
 import { Message } from 'app/services/Apps/Messages/Message';
+import { Tooltip } from 'antd';
 
-type Props = {
+type PropsType = {
   message: Message;
   collectionKey: string;
 };
 
-export default (props: Props) => {
+type ReactionType = { name: string; count: number; users: string[] };
+
+export default ({ message, collectionKey }: PropsType) => {
   const has_reactions =
-    !props.message?.hidden_data?.disable_reactions &&
-    props.message?.reactions &&
-    props.message?.reactions.length > 0;
-  if (!has_reactions) {
-    return <span />;
-  }
+    !message?.hidden_data?.disable_reactions && message?.reactions && message?.reactions.length > 0;
+
+  if (!has_reactions) return <></>;
+
+  const getReactionTooltip = (users: ReactionType['users']): JSX.Element[] =>
+    users.map(id => {
+      const user = Collections.get('users').find(id);
+
+      if (!user) return <></>;
+
+      const name = User.getFullName(user);
+
+      return (
+        <div key={id} style={{ whiteSpace: 'nowrap' }}>
+          {name}
+        </div>
+      );
+    });
+
+  const setReactionClassname = (reaction: ReactionType): string => {
+    const isSelected =
+      message?._user_reaction === reaction.name || reaction.users.includes(User.getCurrentUserId());
+
+    return `reaction ${isSelected ? 'is_selected' : ''}`;
+  };
```

Comment:
I added the `classnames` dependencies, search for `import classNames from "classnames";` to see how to use it

Reply:
Done